sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -168,6 +168,43 @@ case class AlterTableRenameCommand(
 }
 
 /**
+ * A command that add columns to a table
+ * The syntax of using this command in SQL is:
+ * {{{
+ *   ALTER TABLE table_identifier
+ *   ADD COLUMNS (col_name data_type [COMMENT col_comment], ...);
+ * }}}
+*/
+case class AlterTableAddColumnsCommand(
+    table: TableIdentifier,
+    columns: Seq[StructField]) extends RunnableCommand {
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val catalog = sparkSession.sessionState.catalog
+    val catalogTable = DDLUtils.verifyAlterTableAddColumn(catalog, table)
+
+    // If an exception is thrown here we can just assume the table is uncached;
+    // this can happen with Hive tables when the underlying catalog is in-memory.
+    val wasCached = Try(sparkSession.catalog.isCached(table.unquotedString)).getOrElse(false)
```

Comment:
`AlterTableRenameCommand` has similar way to do the uncaching. I thought there might be a reason it exists there. So I did the same. But looking  at the code, it seems you are right. Thanks!